### PR TITLE
haxe: fix head build

### PR DESCRIPTION
**Pull request by ilovezfs**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Supersedes #16672.

### CODE REVIEW

File: Formula/haxe.rb [hunk 1]
```diff
@@ -16,6 +16,13 @@ class Haxe < Formula
   head do
     url "https://github.com/HaxeFoundation/haxe.git", :branch => "development"
     depends_on "opam" => :build
+
+    # ptmap 2.0.2 is needed for OCaml 4.05.0 compatibility
+    # See https://github.com/backtracking/ptmap/pull/4
+    resource "ptmap" do
+      url "https://github.com/ocaml/opam-repository/pull/10170.diff?full_index=1"
+      sha256 "2d5394638c370e654ee1b87e23a348d512da717df9bd52a16c1816c6047ba4f7"
```
Comment (tanis2000) [inline Formula/haxe.rb]: @ilovezfs it looks like the sha256 of 10170.diff is wrong.
When running `brew install haxe --HEAD` I get the following error:

```
==> Downloading https://github.com/ocaml/opam-repository/pull/10170.diff?full_index=1
==> Downloading from https://patch-diff.githubusercontent.com/raw/ocaml/opam-repository/pull/10170.diff?full_index=1
######################################################################## 100,0%
Error: SHA256 mismatch
Expected: 2d5394638c370e654ee1b87e23a348d512da717df9bd52a16c1816c6047ba4f7
Actual: 8b4938166ef02e3546898fb9d5742e466e7cb75f0a9a5548a999fbe38f504628
Archive: /Users/tanis/Library/Caches/Homebrew/haxe--ptmap-10170.diff
To retry an incomplete download, remove the file above.
```

Comment (ilovezfs) [reply Formula/haxe.rb]: Yeah the PR was just updated. I will fix it :)

Comment (ilovezfs) [reply Formula/haxe.rb]: @tanis2000 it should be fixed now via https://github.com/Homebrew/homebrew-core/pull/17316.

```
brew update
rm /Users/tanis/Library/Caches/Homebrew/haxe--ptmap-10170.diff
```
and then try again.

Comment (tanis2000) [reply Formula/haxe.rb]: It's working fine now. Thanks!

Comment (ilovezfs) [reply Formula/haxe.rb]: You're welcome!

**Pull request merged**